zebra-network/src/peer_set/candidate_set/tests/prop.rs

```diff
@@ -33,17 +49,50 @@ proptest! {
             prop_assert![peer.get_last_seen() <= last_seen_limit];
         }
     }
+
+    /// Test that the outbound peer connection rate limit is only applied when
+    /// a peer address is actually returned.
+    ///
+    /// TODO: after PR #2275 merges, add a similar proptest,
+    ///       using a "not ready for attempt" peer generation strategy
+    #[test]
+    fn skipping_outbound_peer_connection_skips_rate_limit(next_peer_attempts in 0..TEST_ADDRESSES) {
```

Comment:
The `skipping_outbound_peer_connection_skips_rate_limit` test checks the bug fix in the code.

(The other test is useful for making sure we're not delaying way too long when we actually have peers, but it can't be too strict due to OS scheduling delays.)